### PR TITLE
Fix nova module cooperation with python-novaclient

### PR DESCRIPTION
**Pull request by slivik**
### What does this PR do?

Fixes `nova` module and `salt.utils.openstack.nova` module:
- fixed Keystone API v3 cooperation
- added Keystone API version detection (based on HTTP status 301 - multiple choices)
- removed novaclient max version restriction

### Previous Behavior
It was not possible to use `nova` module when `python-novaclient > 6.0.1` installed.

### New Behavior
Now `nova` module works with newer version of `python-novaclient`.

### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by gtmanfred — APPROVED** [review]
This looks great, thanks for the help here.

We have also moved everything over to the newer `openstack` shade module starting in 2018.3. but this would be good to backport to 2017.7

**Comment by rallytime** [comment]
@slivik The pylint check found some errors. Can you fix those?

File: salt/utils/openstack/nova.py [hunk 1]
```diff
@@ -64,20 +65,25 @@
...
             return HAS_NOVA
-        elif novaclient_ver > max_ver:
-            log.debug('Older novaclient version required. Maximum: %s',
-                      NOVACLIENT_MAXVER)
-            return False
         log.debug('Newer novaclient version required.  Minimum: %s',
                   NOVACLIENT_MINVER)
     return False
 
+if check_nova:
```
Comment (cachedout) [inline salt/utils/openstack/nova.py]: Did you mean `if check_nova():`?

Comment (slivik) [reply salt/utils/openstack/nova.py]: Sure. Sorry. Once more :)

**Comment by rallytime** [comment]
@slivik This won't back-port cleanly to the 2017.7 branch. Would you be willing to make this change there as well?

**Comment by slivik** [comment]
> @slivik This won't back-port cleanly to the 2017.7 branch. Would you be willing to make this change there as well?

I have created new PR adding openstack nova module and fixing possibility to back-port it cleanly to the 2017.7 branch.

**Pull request merged**